sdk/appconfiguration/azappconfig/client.go

```diff
@@ -0,0 +1,443 @@
+	// The setting versions are the same if their ETag fields match.
+	OnlyIfUnchanged bool
+}
+
+// SetSetting creates a configuration setting if it doesn't exist or overwrites the existing setting in the configuration store.
+func (c *Client) SetSetting(ctx context.Context, setting Setting, options *SetSettingOptions) (SetSettingResult, error) {
+	var ifMatch *azcore.ETag
+	if options != nil && options.OnlyIfUnchanged {
+		ifMatch = setting.ETag
+	}
+
+	resp, err := c.appConfigClient.PutKeyValue(ctx, *setting.Key, setting.toGeneratedPutOptions(ifMatch, nil))
+	if err != nil {
+		return SetSettingResult{}, err
+	}
+
+	return (SetSettingResult)(fromGeneratedSet(resp)), nil
+}
+
+// ListRevisionsPage contains the configuration settings returned by ListRevisionsPager.
+type ListRevisionsPage struct {
+	// Contains the configuration settings returned that match the setting selector provided.
+	Settings []Setting
+
+	// Sync token for the Azure App Configuration client, corresponding to the current state of the client.
+	SyncToken *string
+}
+
+func fromGeneratedGetRevisionsPage(g generated.AzureAppConfigurationClientGetRevisionsResponse) ListRevisionsPage {
+	var css []Setting
+	for _, cs := range g.Items {
+		if cs != nil {
+			css = append(css, settingFromGenerated(*cs))
+		}
+	}
+
+	return ListRevisionsPage{
+		Settings:  css,
+		SyncToken: g.SyncToken,
+	}
+}
+
+// ListRevisionsPager is a Pager for revision list operations.
+//
+// NextPage should be called first. It fetches the next available page of results from the service.
+// If the fetched page contains results, the return value is true, else false.
+// Results fetched from the service can be evaluated by calling PageResponse on this Pager.
+// If the result is false, the value of Err() will indicate if an error occurred.
+type ListRevisionsPager interface {
+	// PageResponse returns the current ListRevisionsPage.
+	PageResponse() ListRevisionsPage
+
+	// Err returns an error if there was an error on the last request.
+	Err() error
+
+	// NextPage returns true if there is another page of data available, false if not.
+	NextPage(context.Context) bool
+}
+
+type listRevisionsPager struct {
+	genPager *generated.AzureAppConfigurationClientGetRevisionsPager
+}
+
+func (p listRevisionsPager) PageResponse() ListRevisionsPage {
+	return fromGeneratedGetRevisionsPage(p.genPager.PageResponse())
+}
+
+func (p listRevisionsPager) Err() error {
+	return p.genPager.Err()
+}
+
+func (p listRevisionsPager) NextPage(ctx context.Context) bool {
+	return p.genPager.NextPage(ctx)
+}
+
+// ListRevisionsOptions contains the optional parameters for the ListRevisions method.
+type ListRevisionsOptions struct {
+}
+
+// NewListRevisionsPager retrieves the revisions of one or more configuration setting entities that match the specified setting selector.
+func (c *Client) NewListRevisionsPager(selector SettingSelector, options *ListRevisionsOptions) ListRevisionsPager {
+	_ = options
```

Comment:
Do you need this for the linter? I think you can ignore it and you should be okay, just a nit.

Reply:
I believe we reverted back to naming the pageable operations just `ListFoo`.